GlotPress/GlotPress-WP
# Add a Grunt action to set GP_SCRIPT_DEBUG to false when we deploy to wporg

## Problem

When we deploy a new version to wporg, we need to minify the CSS and the JavaScript tiles. 

- In https://github.com/GlotPress/GlotPress/pull/1614 we added a `GP_SCRIPT_DEBUG` constant to load the unminified CSS and JavaScript files by default, setting this `GP_SCRIPT_DEBUG` constant to `true` in the **glotpress.php** file.
- In https://github.com/GlotPress/GlotPress/pull/1654 and in https://github.com/GlotPress/GlotPress/pull/1655 we added some actions to minify this files when we deploy to wporg.

But we need to change the `GP_SCRIPT_DEBUG` constant to `false` before we deploy to wporg, so the wporg release will use the minified files (except the end user will use the `SCRIPT_DEBUG` set to `true`). This PR address this problem.

Fixes https://github.com/GlotPress/GlotPress/issues/1587.



## Solution

This PR adds:
- A **Grunt action** to change the `define( 'GP_SCRIPT_DEBUG', true );` to `define( 'GP_SCRIPT_DEBUG', false );` in the glotpress.php file. **Gruntfile.js** file.
- A **npm action** to run the Grunt action. **package.json** file.
- This npm action into the **wporg workflow**. **.github/workflows/deploy-wporg-plugin.yml** file.



## Testing Instructions

To test this PR:
- You can execute the Grunt action with the `grunt replace:updateGPScriptDebugConst` command.
- You can execute the npm action with the `npm run updateGPScriptDebugConst` command.

In both situations, you can check that the `GP_SCRIPT_DEBUG` value in the `glotpress.php` file has changed from true to false.
